.github/workflows/build.yml

```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9"]
+        python-version: ["3.10", "3.11", "3.12"]
```

Comment:
I think that the test failures in the CI seem to be related to the changes in these versions. I ran into a similar issue previously, so I think it's worth taking a look at the tests...